include/eve/detail/function/simd/riscv/bit_cast.hpp

```diff
@@ -0,0 +1,263 @@
+    else if constexpr( out_lmul == 4 ) return __riscv_vreinterpret_u32m4(x);
+    else if constexpr( out_lmul == 8 ) return __riscv_vreinterpret_u32m8(x);
+  }
+  else if constexpr( match(out_c, category::int16) )
+  {
+    if constexpr( out_lmul == -4 ) return __riscv_vreinterpret_i16mf4(x);
+    else if constexpr( out_lmul == -2 ) return __riscv_vreinterpret_i16mf2(x);
+    else if constexpr( out_lmul == 1 ) return __riscv_vreinterpret_i16m1(x);
+    else if constexpr( out_lmul == 2 ) return __riscv_vreinterpret_i16m2(x);
+    else if constexpr( out_lmul == 4 ) return __riscv_vreinterpret_i16m4(x);
+    else if constexpr( out_lmul == 8 ) return __riscv_vreinterpret_i16m8(x);
+  }
+  else if constexpr( match(out_c, category::uint16) )
+  {
+    if constexpr( out_lmul == -4 ) return __riscv_vreinterpret_u16mf4(x);
+    else if constexpr( out_lmul == -2 ) return __riscv_vreinterpret_u16mf2(x);
+    else if constexpr( out_lmul == 1 ) return __riscv_vreinterpret_u16m1(x);
+    else if constexpr( out_lmul == 2 ) return __riscv_vreinterpret_u16m2(x);
+    else if constexpr( out_lmul == 4 ) return __riscv_vreinterpret_u16m4(x);
+    else if constexpr( out_lmul == 8 ) return __riscv_vreinterpret_u16m8(x);
+  }
+  else if constexpr( match(out_c, category::int8) )
+  {
+    if constexpr( out_lmul == -8 ) return __riscv_vreinterpret_i8mf8(x);
+    else if constexpr( out_lmul == -4 ) return __riscv_vreinterpret_i8mf4(x);
+    else if constexpr( out_lmul == -2 ) return __riscv_vreinterpret_i8mf2(x);
+    else if constexpr( out_lmul == 1 ) return __riscv_vreinterpret_i8m1(x);
+    else if constexpr( out_lmul == 2 ) return __riscv_vreinterpret_i8m2(x);
+    else if constexpr( out_lmul == 4 ) return __riscv_vreinterpret_i8m4(x);
+    else if constexpr( out_lmul == 8 ) return __riscv_vreinterpret_i8m8(x);
+  }
+  else if constexpr( match(out_c, category::uint8) )
+  {
+    if constexpr( out_lmul == -8 ) return __riscv_vreinterpret_u8mf8(x);
+    else if constexpr( out_lmul == -4 ) return __riscv_vreinterpret_u8mf4(x);
+    else if constexpr( out_lmul == -2 ) return __riscv_vreinterpret_u8mf2(x);
+    else if constexpr( out_lmul == 1 ) return __riscv_vreinterpret_u8m1(x);
+    else if constexpr( out_lmul == 2 ) return __riscv_vreinterpret_u8m2(x);
+    else if constexpr( out_lmul == 4 ) return __riscv_vreinterpret_u8m4(x);
+    else if constexpr( out_lmul == 8 ) return __riscv_vreinterpret_u8m8(x);
+  }
+}
+
+// change sew only.
+// For example:
+// unsigned -> unsigned char
+// int -> char
+// lmul can not be changed.
+// doc. line:
+// Reinterpret between different SEW under the same LMUL
+template<callable_options O, scalar_value T, typename N, scalar_value U, typename M>
+EVE_FORCEINLINE wide<U, M>
+bit_cast_(EVE_REQUIRES(rvv_), const O&, wide<T, N> x, as<wide<U, M>> const&) noexcept
+requires rvv_abi<abi_t<T, N>> && rvv_abi<abi_t<U, M>> && (sizeof(T) != sizeof(U))
+         && (have_common_category<wide<T, N>, wide<U, M>>(category::int_,
```

Comment:
what is the motivation for this particular requires ?

Reply:
Well, original idea was to have the exact match with RISC-V intrinsics documentation. For `vreinterpret` intrinsics we have 3 main categories:
1) the one that change SEW, but not change type characteristics (so you can cast `int` to `char`, but not `char` to `float`)
2) that change type characteristics, but not SEW (casts from `float` to `int`)
3) casts from mast to m1 vector registers - they should go to `simd_cast` (well, for now I use them as `rvv_simd_cast`, so not available in normal `simd_cast`). The reason to go to `simd_cast` is because `sizeof()` for mask types defined quite bad, so it is not correspond to real layout of mask register.

Anyway, for now removed this